docs/compose-file.md

```diff
@@ -428,9 +428,17 @@ Both `$VARIABLE` and `${VARIABLE}` syntax are supported. Extended shell-style
 features, such as `${VARIABLE-default}` and `${VARIABLE/foo/bar}`, are not
 supported.
 
-If you need to put a literal dollar sign in a configuration value, use a double
-dollar sign (`$$`).
+When you want to set an environment variable that is not to be interpolated by
+Compose, or the configuration value needs a literal dollar sign you can use a
+double dollar sign (`$$`).
 
```

Comment:
You can use a `$$` (double-dollar sign) when your configuration needs a literal dollar sign.  This also prevents Compose from interpolating a value, so `$$` useful for referring to environment variables but disabling processing by Compose.

<snip>

If you forget an use a single `$` Compose interprets the value as an environment variable and warns:

<snip>